pkg/collector/process/process.go

```diff
@@ -491,8 +524,48 @@ func (c *collector) Collect(ctx *types.ScrapeContext, ch chan<- prometheus.Metri
 			processName,
 			pid,
 			cpid,
+			processMap[pid],
 		)
 	}
 
 	return nil
 }
+
+func getUserProcess() map[string]string {
+	processMap := make(map[string]string)
+
+	var users []Win32_LoggedOnUser
+	var sessionProcesses []Win32_SessionProcess
+
+	err := wmi.Query(win32LoggedOnUserQuery, &users)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	err = wmi.Query(win32SessionProcessQuery, &sessionProcesses)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	for _, sesessionProcess := range sessionProcesses {
+		for _, user := range users {
+			if user.Dependent == sesessionProcess.Antecedent {
+				re := regexp.MustCompile(`Domain="(.+?)",Name="(.+?)"`)
+				match := re.FindStringSubmatch(user.Antecedent)
+
+				domain := match[1]
+				name := match[2]
+				fullname := fmt.Sprintf("%s\\%s", domain, name)
+
+				reps := regexp.MustCompile(`Handle="(\d+)"`)
```

Comment:
could you put this to the global scope to avoid that the regex is evaluated on each loop?

Reply:
Hello, done with new commit